hub/src/server/server.ts

```diff
@@ -73,6 +73,51 @@ export class HubServer {
     }
   }
 
+  async handleDelete(
+    address: string, path: string,
+    requestHeaders: { authorization?: string }
+  ) {
+    const oldestValidTokenTimestamp = await this.authTimestampCache.getAuthTimestamp(address)
+    this.validate(address, requestHeaders, oldestValidTokenTimestamp)
+
+    // can the caller delete? if so, in what paths?
+    const scopes = getAuthenticationScopes(requestHeaders.authorization)
+    const deletePrefixes = []
+    const deletePaths = []
+    for (let i = 0; i < scopes.length; i++) {
+      if (scopes[i].scope == 'deleteFilePrefix') {
+        deletePrefixes.push(scopes[i].domain)
+      } else if (scopes[i].scope == 'deleteFile') {
+        deletePaths.push(scopes[i].domain)
+      }
+    }
+
+    if (deletePrefixes.length > 0 || deletePaths.length > 0) {
```

Comment:
This is fine for now, but before we implement something else that's going to add new scopes (say "append" with appendPaths and appendPrefixes), we would probably want to factor this and the putFile scopes check so that we don't have a bunch of duplicated logic.